File: 1. Two Sum/1. Two Sum.py
#
# Problem: 1. Two Sum
# Difficulty: Easy
# Link: https://leetcode.com/problems/two-sum/
# Language: python3
# Date: 2026-03-25


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hm = {}
        for i in range(len(nums)):
            if nums[i] in hm:
                return [hm[nums[i]], i]

            hm[target - nums[i]] = i
        return [-1, -1]

File: 1. Two Sum/1. Two Sum.swift
/*
 * Problem: 1. Two Sum
 * Difficulty: Easy
 * Link: https://leetcode.com/problems/two-sum/
 * Language: swift
 * Date: 2026-05-10
 */

class Solution {
    func twoSum(_ nums: [Int], _ target: Int) -> [Int] {
        var map: [Int: Int] = [:]
        for i in 0...nums.count {
            if map[target-nums[i]] != nil {
                return [map[target-nums[i]] ?? i, i]
            }
            map[nums[i]]=i
        }
        return []
    }
}
